atomic_reactor/utils/resource_manager.py

```diff
@@ -67,21 +71,62 @@ def username(self):
     def ssh_keyfile(self):
         return self._ssh_keyfile
 
+    @property
+    def slots(self):
+        return self._slots
+
+    @slots.setter
+    def slots(self, num: int):
+        self._slots = num
+
+    @property
+    def slots_dir(self):
+        if self._slots_dir is None:
+            with self._ssh_session() as client:
+                home, _, _ = self._ssh_run_remote_cmd(client, "pwd")
+            self._slots_dir = os.path.join(home, SLOTS_RELATIVE_PATH)
+        return self._slots_dir
+
+    def get_slot_path(self, slot_id: int):
+        return os.path.join(self.slots_dir, f"slot_{slot_id}.json")
+
+    def _ensure_slot_exists(self, slot_id: int):
+        """ Create the slot file (with '{}' in content) if it's missing on host """
+        slot_path = self.get_slot_path(slot_id)
+        _cmd = f"test -s {slot_path} || echo {{}} > {slot_path}"
+        cmd = f"flock {slot_path} -c {quote(_cmd)}"
+        with self._ssh_session() as client:
+            self._ssh_run_remote_cmd(client, cmd)
+
+    def read_slot(self, slot_id: int):
+        """
+        Read slot content and JSON deserialize it to a Python object
+
+        Note: when slot file does not exists, calling this API will also create
+        the slot file with "{}" in content.
+        """
+        if slot_id not in range(self.slots):
+            raise RuntimeError(f"Slot id {slot_id} exceeds the max available slots {self.slots}")
+
+        slot_path = self.get_slot_path(slot_id)
+        self._ensure_slot_exists(slot_id)
+
+        with self._ssh_session() as client:
+            cmd = self.get_cmd_to_read_from_file(slot_path)
+            return self._ssh_run_remote_cmd_to_get_json(client, cmd)
+
     def _validate_resource_lock(self, payload: dict):
         """ Validate resource lock json file against schema """
-        with open(RESOURCE_LOCK_SCHEMA, 'r') as f:
-            schema_data = f.read()
-        schema = json.loads(schema_data)
         try:
-            jsonschema.validate(payload, schema)
-        except jsonschema.ValidationError as err:
-            logger.warning("Invalid json file: %s", err)
+            validate_with_schema(payload, RESOURCE_LOCK_SCHEMA)
+        except OsbsValidationException as e:
+            logger.warning("Invalid json file: %s", str(e))
```

Comment:
just nitpick, but isn't `%s` in format string doing `str()` call? So `str(e)` is duplicated